In [81]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
%matplotlib inline

## Ребров. Домашка. Урок 4
Спасибо, что рассказали как пользоваться жупитером! это оказалось полезно и очень удобно. я пока не освоил латех, поэтому записываю как умею, постараюсь освоить и латех поскорее.
### Задание 1
D(x) = 0.2  
a = 0.5


In [82]:
D = 0.2
a = 0.5
b = np.sqrt(D * 12) + a
srednee = (a+b)/2
print(f'{b} - правая граница')
print(f'{srednee} - среднее значение')

2.049193338482967 - правая граница
1.2745966692414834 - среднее значение


### Задание 2
h = 190  
M = 178  
D = 25  
Правильно ли я понимаю, что по условию надо узнать, сколько сигм в (190-178)?


In [83]:
h = 190
m = 178
d = 25
std = np.sqrt(d)
otvet = (190-178)/std
otvet

2.4

### Задание 3  
я не совсем понял как с помощью методички делать это задание, поэтому воспользовался таблицей значений функции Лапласа, вроде бы это не противоречит условию не пользоваться питоновскими библиотеками. Кроме того, в пунктах, где нет точных границ, я использовал 0 и 250 см как крайние значения, хотя 250 и явно перебор. Ну и так как кодом там считать практически нечего, я с его помощью добывал только значения функции Ф, вычитал, складывал и делил.
m = 174  
std = 8  
p = ?  
1. больше 182 см?  
|  Ответ: P1 = 0.1291   
2. больше 190 см?  
|  Ответ: P2 = 0.0226   
3. от 166 см до 190 см?  
|  Ответ: P3 = 0.8185  
4. от 166 см до 182 см?  
|  Ответ: P4 = 0.6826  
5. от 158 см до 190 см?  
|  Ответ: P5 = 0.9544  
6. не выше 150 см или не ниже 190 см?  
|  Ответ: P6 = 0.0240   
7. не выше 150 см или не ниже 198 см?  
|  Ответ: P7 = 0.0026  
8. ниже 166 см?  
|  Ответ: P8 = 0.1291  

In [84]:
#Обозначаю переменные промежутков роста по условиям задачи
std = 8
m = 174
a1 = 183
b1 = 250
a2 = 191
b2 = 250
a3 = 166
b3 = 190
a4 = 166
b4 = 182
a5 = 158
b5 = 190
a61 = 0
b61 = 150
a62 = 190
b62 = 250
a71 = 0
b71 = 150
a72 = 198
b72 = 250
a8 = 0
b8 = 165


In [85]:
#Считаю значения t  для подстановки в F(t)
#1
f11 = (b1 - m)/std
f12 = (a1 - m)/std
#2
f21 = (b2 - m)/std
f22 = (a2 - m)/std
#3
f31 = (b3 - m)/std
f32 = (a3 - m)/std
#4
f41 = (b4 - m)/std
f42 = (a4 - m)/std
#5
f51 = (b5 - m)/std
f52 = (a5 - m)/std
#6.1
f611 = (b61 - m)/std
f612 = (a61 - m)/std
#6.2
f621 = (b62 - m)/std
f622 = (a62 - m)/std
#7.1
f711 = (b71 - m)/std
f712 = (a71 - m)/std
#7.2
f721 = (b72 - m)/std
f722 = (a72 - m)/std
#8
f81 = (b8 - m)/std
f82 = (a8 - m)/std
print(f'1: {f11} - {f12}\n2: {f21} - {f22}\n3: {f31} - {f32}\n4: {f41} - {f42}\n5: {f51} - {f52}\n6.1: {f611} - {f612}; 6.2: {f621} - {f622}\n7.1: {f711} - {f712}; 7.2: {f721} - {f722}\n8: {f81} - {f82}\n')

1: 9.5 - 1.125
2: 9.5 - 2.125
3: 2.0 - -1.0
4: 1.0 - -1.0
5: 2.0 - -2.0
6.1: -3.0 - -21.75; 6.2: 9.5 - 2.0
7.1: -3.0 - -21.75; 7.2: 9.5 - 3.0
8: -1.125 - -21.75



Я предположил, что "больше 182" означает 183 и выше, из-за этого табличные значения не так красивы как могли бы быть.
значение Ф:  
1: для 9.5 = 0.4999, для 1.12 = 0.3708  
2: для 9.5 = 0.4999, для 2.0 = 0.4772  
3: для 2.0 = 0.4772, для -1.0 = -0.3413   
4: для 1.0 = 0.3413 , для -1.0 = -0.3413    
5: для 2.0 = 0.4772, для -2.0 = -0.4772   
6: для -3.0 = -0.4986 , для -21.75 = -0.4999 | для 9.5 = 0.4999, для 2.0 = 0.4772  
7: для -3.0 = -0.4986 , для -21.75 = -0.4999 | для 9.5 = 0.4999, для 3.0 = 0.4986    
8: для -1.12 = -0.3708, для -21.75 = -0.4999



In [86]:
#Подставляю табличные значения в функции Ф
F11 = 0.4999
F12 = 0.3708
F21 = 0.4999
F22 = 0.4772
F31 = 0.4772
F32 = -0.3413
F41 = 0.3413
F42 = -0.3413
F51 = 0.4772
F52 = -0.4772
F611 = -0.4986
F612 = -0.4999
F621 = 0.4999
F622 = 0.4772
F711 = -0.4986
F712 = -0.4999
F721 = 0.4999
F722 = 0.4986 
F81 = -0.3708
F82 = -0.4999


In [88]:
#Считаю вероятности
P1 = F11 - F12
P2 = F21 - F22
P3 = F31 - F32
P4 = F41 - F42
P5 = F51 - F52
P6 = (F611 - F612) + (F621 - F622)
P7 = (F711 - F712) + (F721 - F722)
P8 = F81 - F82
print(f'P1: {P1}\nP2: {P2}\nP3: {P3}\nP4: {P4}\nP5: {P5}\nP6: {P6}\nP7: {P7}\nP8: {P8}')

P1: 0.1291
P2: 0.022699999999999998
P3: 0.8185
P4: 0.6826
P5: 0.9544
P6: 0.02400000000000002
P7: 0.0026000000000000467
P8: 0.1291


### Ответы на задание 3:  
1. больше 182 см?  
|  Ответ: P1 = 0.1291   
2. больше 190 см?  
|  Ответ: P2 = 0.0226   
3. от 166 см до 190 см?  
|  Ответ: P3 = 0.8185  
4. от 166 см до 182 см?  
|  Ответ: P4 = 0.6826  
5. от 158 см до 190 см?  
|  Ответ: P5 = 0.9544  
6. не выше 150 см или не ниже 190 см?  
|  Ответ: P6 = 0.0240   
7. не выше 150 см или не ниже 198 см?  
|  Ответ: P7 = 0.0026  
8. ниже 166 см?  
|  Ответ: P8 = 0.1291  